demo/api/schema.gql

```diff
@@ -562,13 +562,13 @@ type Mutation {
   updateRedirectActiveness(id: ID!, input: RedirectUpdateActivenessInput!): Redirect!
   deleteRedirect(id: ID!): Boolean!
   updateDamFile(id: ID!, input: UpdateDamFileInput!): DamFile!
-  moveDamFiles(fileIds: [ID!]!, targetFolderId: ID): [DamFile!]!
+  moveDamFiles(fileIds: [ID!]!, targetFolderId: ID, scope: DamScopeInput!): [DamFile!]!
```

Comment:
if moving between scopes is not supported, why do we need a scope arg?

Reply:
copying files and folders into another scope will be an important feature we need. How is that planned?

Reply:
Good question! I thought about the following scenario: The mutation is called without `targetFolderId` (= a move to the root folder), so I'm not able to derive the target scope from the target folder. Theoretically, it might be possible to move multiple files in different scopes to the respective root folders. Maybe this is okay

One benefit of having the scope arg is that the scope guard works out of the box. Otherwise, we need to provide a way to determine the scopes of multiple entities (which we'll have to do someday anyway, but we don't have a solution right now). Should I remove the scope arg?

I definitely need it for moveDamFolders though

Reply:
> copying files and folders into another scope will be an important feature we need. How is that planned?

As far as I understood, this should be done via sharing the files with another scope (not actually copying).

Reply:
To my understanding, a file should never be able to change its scope, right? So it would also be possible to omit the scope and keep the current scope of the file

Reply:
> As far as I understood, this should be done via sharing the files with another scope (not actually copying).

yep, true

Reply:
> To my understanding, a file should never be able to change its scope, right? So it would also be possible to omit the scope and keep the current scope of the file

Wouldn't this be somehow arbitrary? I may call the mutation without `targetFolderId` and move multiple files in different scopes to the respective root folders. Are we okay with that?

Reply:
> you still have to make sure the scope of the source and target equals that scope....

Without the scope arg, the mutation would do the following

1. If there's no `targetFolderId`, move all files to the root in the respective scopes (no validation of scope match)
2. If there's a `targetFolderId`, use the scope of folder to validate if source files and target folder match.

Do we prefer this?

Reply:
> Wouldn't this be somehow arbitrary? I may call the mutation without `targetFolderId` and move multiple files in different scopes to the respective root folders. Are we okay with that?

Yes, you are right. Then I think we should keep the scope arg

Reply:
> I may call the mutation without targetFolderId and move multiple files in different scopes to the respective root folders. Are we okay with that?

no, we have to validate that moving files doesn't change their scope.

Reply:
I've decided to remove the scope argument: c14ed628759be92c2ce20f55e292345a32f9285b. "Misusing" the argument for access validation is not a good idea.

> But can the situation you describe occur naturally?

No, it won't. But just because we're not creating such a situation in the admin, it doesn't mean we shouldn't think about it. We should always expect a nifty user to directly talk to our GraphQL API.

However, while the mutation may be arbitrary (and will rarely occur), the result is still valid. All files will be moved to the root folder in their own scope, but no file will be moved from one scope to another

> no, we have to validate that moving files doesn't change their scope.

I've added a files scope <-> target folder scope validation if a target folder is present.

One remaining TODO is to validate general files and folder access. Should I do this in the resolver, or in the service?

Reply:
> Should I do this in the resolver, or in the service?

haha :) whatever fits best....

Reply:
Done: e4c88ff6726196443c4f7b2537b17234dfc81c52